From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Abel <abelvanhulst@gmail.com>
Date: Wed, 18 Dec 2024 13:14:06 +0100
Subject: [PATCH] Support all versions


diff --git a/src/main/java/pathfinding/platform/Location.java b/src/main/java/pathfinding/platform/Location.java
index ae6c69c5bb04231d1fda7dec2df88cd9bd30e203..9dad1b7a4e20838570f28d0e15e60effbcef5967 100644
--- a/src/main/java/pathfinding/platform/Location.java
+++ b/src/main/java/pathfinding/platform/Location.java
@@ -119,7 +119,12 @@ public class Location implements Cloneable {
 
     public boolean isSolidBlock() {
 
-        return Block.isShapeFullBlock(this.getBLockState().getOcclusionShape());
+        return Block.isShapeFullBlock(this.getBLockState().getOcclusionShape(
+                #if MC_VER <= MC_1_21_0
+                level,
+                new BlockPos(getBlockX(), getBlockY(), getBlockZ())
+                #endif
+        ));
 
     }
 
